src/utils/daos_hdlr.c

```diff
@@ -1535,6 +1536,758 @@ cont_destroy_hdlr(struct cmd_args_s *ap)
+			DAOS_PC_RW, &ap->dst_pool, NULL, NULL);
+			if (rc != 0) {
+				fprintf(stderr, "failed to connect to destination pool: %d\n", rc);
+			}
+		if (daos_uuid_valid(ap->dst_c_uuid)) { 
+			rc = daos_cont_open(ap->dst_pool, ap->dst_c_uuid, DAOS_COO_RW,
+				&ap->dst_cont, dst_cont_info, NULL);
+			rc = dfs_mount(ap->dst_pool, ap->dst_cont, O_RDWR, &(daos_dst_file->dfs));
+			if (rc) {
+				fprintf(stderr, "dfs mount on destination failed: %s, %d\n",
+					ap->dst_path, rc);
+			}
+		}
+	} else {
+		/* Resolve pool, container UUIDs from dst path if needed */
+        	if (ap->dst_path != NULL) {
+			rc = duns_resolve_path(ap->dst_path, dst_dattr);
+			if (rc == 0) { 
+				ap->type = (*dst_dattr).da_type;
+				uuid_copy(ap->dst_p_uuid, (*dst_dattr).da_puuid);
+				uuid_copy(ap->dst_c_uuid, (*dst_dattr).da_cuuid);
+				rc = daos_pool_connect(ap->dst_p_uuid, ap->sysname, ap->dst_mdsrv,
+						DAOS_PC_RW, &ap->dst_pool, NULL, NULL);
+				if (rc != 0) {
+					fprintf(stderr, "failed to connect to destination pool: %d\n", rc);
+				}
+				rc = daos_cont_open(ap->dst_pool, ap->dst_c_uuid, DAOS_COO_RW,
+					&ap->dst_cont, dst_cont_info, NULL);
+				rc = dfs_mount(ap->dst_pool, ap->dst_cont, O_RDWR, &(daos_dst_file->dfs));
+				if (rc) {
+					fprintf(stderr, "dfs mount on destination failed: %s, %d\n",
+						ap->dst_path, rc);
+				}
+			} else {
+				daos_dst_file->type = POSIX;
+			}
+		}
+	}
+	/* create DAOS POSIX destination container if it doesn't exist */
+	if (daos_dst_file->type == DAOS) {
+		if (daos_uuid_valid(ap->src_c_uuid) && !daos_uuid_valid(ap->dst_c_uuid)) {
+			/* create POSIX destination container if it doesn't exist */
+	    		if (uuid_is_null(ap->c_uuid))
+				uuid_generate(ap->dst_c_uuid);
+			rc = dfs_cont_create(ap->dst_pool, ap->dst_c_uuid, NULL, &(ap->dst_cont),
+						&(daos_dst_file->dfs));
+			if (rc != 0) {
+				fprintf(stderr, "failed to create container: %s (%d)\n",
+					d_errdesc(rc), rc);
+				return rc;
+			}
+			fprintf(stdout, "Successfully created container "DF_UUIDF"\n",
+				DP_UUID(ap->dst_c_uuid));
+		}
+	}
+	return rc;
+}
+
+static void
+daos_file_set_defaults(daos_file_t *daos_file)
+{
+	/* set defaults for daos_file struct */
+	daos_file->type	  = DAOS;
+	daos_file->fd     = -1; 
+	daos_file->offset = 0; 
+	daos_file->obj    = NULL; 
+	daos_file->dfs    = NULL; 
+}
+
+int
+fs_copy_hdlr(struct cmd_args_s *ap)
+{
+	int rc = 0;
+	daos_cont_info_t src_cont_info;
+	daos_cont_info_t dst_cont_info;
+	struct duns_attr_t dst_dattr = {0};
+	struct duns_attr_t src_dattr = {0};
+
+	/* daos_file_t abstraction for POSIX and DFS files */
+	daos_file_t daos_src_file;
+	daos_file_t daos_dst_file;
+	daos_file_set_defaults(&daos_src_file);
+	daos_file_set_defaults(&daos_dst_file);
+	fs_copy_connect(&daos_src_file, &daos_dst_file, ap,
+			   &src_cont_info, &dst_cont_info,
+			   &src_dattr, &dst_dattr);
+
+	/* set paths based on file type for source and destination */
```

Comment:
We don't handle POSIX -> POSIX here, and I imagine that's probably okay, but do we want to handle it, or at least have an `else ... error`?

Reply:
@dsikich Actually, I think we might want to handle POSIX -> POSIX, for the case of DFUSE -> DFUSE

Reply:
@daltonbohning I don't really see why someone would do that? If they are using two dfuse mounted directories why not just use cp?

Reply:
@daltonbohning it is just kind of an odd thing to do, but I guess we can account for this. Right now, I will add a message that says it is not supported now at the very least.

Reply:
@dsikich Yeah, it would be weird, but I imagine if someone _**did**_ try, they might be confused about why they can't do it. Since we support POSIX as the source or as the destination, I don't think it should be difficult to allow that option, right?
But yeah, we don't have to add it right now

Reply:
Also, I think the POSIX->POSIX case would allow us to create a baseline to compare against `cp`. I.e. we can run both `cp` and `daos fs copy` on edge cases to make sure they are consistent. Then, as long as the DFS I/O functions match the POSIX I/O functions, we can know that we're handling these edge cases correctly.
I've actually done this when testing `dcp`.

Reply:
@daltonbohning yeah shouldn't be hard to add